enhancements/network/service-egress-traffic-steering.md

```diff
@@ -63,20 +69,24 @@ The ingress part will be handled by a LoadBalancer provider, such as MetalLB, th
 It is worth noting that in MetalLB's case, a given LoadBalancer service can be announced by multiple L2 and BGP advertisements, possibly being (even accidently) announced from multiple nodes. For our use-case the user MUST take care of configuring their MetalLB resources in a way that the service is announced only by the node holding the label - a full example is detailed in [Usage Example](#Usage-Example).
 
-To achieve these goals, we introduce a new resource `EgressService` for users to create alongside LoadBalancer services which can be either empty or contain optional fields:
-- `nodeSelector`: allows limiting the nodes that can be selected to handle the service's traffic.
+### Implementation Details/Notes/Constraints
+
+A new resource `EgressService` is supported alongside LoadBalancer services with the following optional fields:
+- `snat`: Determines if the source IP of egress traffic originating from the pods backing the Service
+should be set to its ingress IP.
+When the field is not specified the source IP is set according to the interface of the Network, leveraging the masquerade rules that are already in place.
+  - `nodeSelector`: Allows limiting the nodes that can be selected to handle the service's traffic.
 When present only a node whose labels match the specified selectors can be selected for handling the service's traffic as explained earlier.
 When the field is not specified any node in the cluster can be chosen to manage the service's traffic.
 
 - `network`: The network which this service should send egress and corresponding ingress replies to.
 This is typically implemented as VRF mapping, representing a numeric id or string name of a routing table which by omission uses the default host routing.
 
-### Implementation Details/Notes/Constraints
+Although the fields are optional, an empty resource does not have any meaning and would not modify any behavior.
 
-A new resource `EgressService` is supported alongside LoadBalancer services.
-When `ovnkube-master` detects that a LoadBalancer service has a corresponding `EgressService` it will elect a node to act as the point for all of the traffic of that service (ingress/egress). If the resource contains valid LabelSelectors in its `nodeSelector` field only a node whose labels match the selectors can be elected.
+When `ovnkube-master` detects that a LoadBalancer service has a corresponding `EgressService` with `snat` specified it will elect a node to act as the point for all of the traffic of that service (ingress/egress). If the resource contains valid LabelSelectors in its `snat.nodeSelector` field only a node whose labels match the selectors can be elected.
 The specified selectors have to match at least one of the nodes in the cluster, otherwise we don't configure anything.
-If the `nodeSelector` field is not specified any node can be elected.
+If the `snat.nodeSelector` field is not specified any node can be elected.
```

Comment:
wait this is the first time in this document it nodeSelector looks like a sub-field of snat? You should clarify the  list of new fields you added in the above implementation paragraph.

Reply:
sorry, I thought it was clear as a sub-bullet under the snat field. changed it now to snat.nodeSelector there